search-books/src/main/java/nix/education/repository/TagRepository.java

```diff
@@ -0,0 +1,10 @@
+package nix.education.repository;
+
+import nix.education.entity.Tag;
+import org.springframework.data.repository.CrudRepository;
+
+public interface TagRepository extends CrudRepository<Tag, Long> {
+
+    Tag findTagByTag(String tagTitle);
+    boolean  existsTagByTag (String tagTitle);
```

Comment:
This is too complicated, I  guess. I'd rather declare a method similar to:
`List<Tag> findByTagIn(List<String> tags);`

Reply:
Yes, I agree. Already use this method: List<Tag> findTagsByTagIn (List<String> tags);